src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -169,7 +171,17 @@ public int order() {
                 attachSourceFieldContext(request);
             }
             final Set<String> injectedRoles = rolesInjector.injectUserAndRoles(threadContext);
-            final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            boolean enforcePrivilegesEvaluation = false;
+            User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            if(user == null && threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER) != null) {
```

Comment:
```suggestion
            if(user == null) {
```
Will it be better to rely on `backendRegistry.authenticate(request, null, task, action);`?